datastore/prometheus.go

```diff
@@ -146,17 +160,96 @@ func (p *PrometheusExporter) updateMetrics(toPod PodEvent, req Request) {
 	}).Inc()
 }
 
+func (p *PrometheusExporter) handlePackets() {
+	for {
+		select {
+		case <-p.ctx.Done():
+			return
+		case pkt := <-p.pktChanBuffer:
+			p.handlePacket(pkt)
+		}
+	}
+}
+
+func (p *PrometheusExporter) handlePacket(pkt Packet) {
+	labels := prometheus.Labels{
+		"toPort": strconv.Itoa(int(pkt.ToPort)),
+	}
+
+	if pkt.FromType == PodSource {
+		fromPod, found := p.podCache.get(pkt.FromUID)
+		if found {
+			labels["fromPod"] = fromPod.(PodEvent).Name
+			labels["fromAcornApp"] = fromPod.(PodEvent).Labels[appLabel]
+			labels["fromAcornAppNamespace"] = fromPod.(PodEvent).Labels[appNamespaceLabel]
+			labels["fromAcornContainer"] = fromPod.(PodEvent).Labels[containerLabel]
+
+			if pkt.ToType == PodDest {
+				toPod, found := p.podCache.get(pkt.ToUID)
+				if found {
+					labels["toPod"] = toPod.(PodEvent).Name
+					labels["toAcornApp"] = toPod.(PodEvent).Labels[appLabel]
+					labels["toAcornAppNamespace"] = toPod.(PodEvent).Labels[appNamespaceLabel]
+					labels["toAcornContainer"] = toPod.(PodEvent).Labels[containerLabel]
+				}
+			} else if pkt.ToType == OutsideDest {
+				labels["toHostname"] = pkt.ToUID
+			} else if pkt.ToType == ServiceDest {
+				log.Logger.Warn().Msgf("Pod %s in namespace %s sent traffic to service with uid %s)", fromPod.(PodEvent).Name, fromPod.(PodEvent).Namespace, pkt.ToUID)
+				labels["toPod"] = pkt.ToIP
+			} else {
+				labels["toPod"] = pkt.ToIP
+			}
+		}
+	} else if pkt.FromType == OutsideSource {
+		labels["fromHostname"] = pkt.FromUID
+
+		if pkt.ToType == PodDest {
+			toPod, found := p.podCache.get(pkt.ToUID)
+			if found {
+				labels["toPod"] = toPod.(PodEvent).Name
+				labels["toAcornApp"] = toPod.(PodEvent).Labels[appLabel]
+				labels["toAcornAppNamespace"] = toPod.(PodEvent).Labels[appNamespaceLabel]
+				labels["toAcornContainer"] = toPod.(PodEvent).Labels[containerLabel]
+			}
+		} else if pkt.ToType == ServiceDest {
+			log.Logger.Warn().Msgf("Host %s (outside) sent traffic to service with uid %s)", pkt.FromUID, pkt.ToUID)
+			labels["toPod"] = pkt.ToIP
+		} else {
+			labels["toPod"] = pkt.ToIP
+		}
```

Comment:
nit: never miss a chance to switchify

```suggestion
        switch pkt.ToType {
        case PodDest:
            if toPod, found := p.podCache.get(pkt.ToUID); found {
                pe := toPod.(PodEvent)
                labels["toPod"] = pe.Name
				labels["toAcornApp"] = pe.Labels[appLabel]
				labels["toAcornAppNamespace"] = pe.Labels[appNamespaceLabel]
				labels["toAcornContainer"] = pe.Labels[containerLabel]
            }
        case ServiceDest:
            log.Logger.Warn().Msgf("Host %s (outside) sent traffic to service with uid %s)", pkt.FromUID, pkt.ToUID)
            fallthrough
        default:
            labels["toPod"] = pkt.ToIP
        }
```